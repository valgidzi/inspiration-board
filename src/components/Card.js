import React from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

const Card = (props) => {

  const cardText = props.text

  const cardEmoji = props.emoji ? emoji.getUnicode(props.emoji) : ''

  const onDeleteClick = () => {
    props.deleteCardCallback(props.id);
  }

  return (
    <section className="card">
      <div className="card__content">
        <div className="card__content-text">
          {cardText}
        </div>
        <div className="card__content-emoji">
          {cardEmoji}
        </div>
        <button
          className="card__delete"
          onClick={onDeleteClick}>
          DISCARD
        </button>
      </div>
    </section>
  )
}

Card.propTypes = {
  id: PropTypes.number,
  text: PropTypes.string,
  emoji: PropTypes.string,
  deleteCardCallback: PropTypes.func
};

export default Card;
