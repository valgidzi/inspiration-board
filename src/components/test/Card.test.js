import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {

    const wrapper = shallow( <Card text="text" emoji="emoji"/>);

    expect(wrapper).toMatchSnapshot();
  });
});
