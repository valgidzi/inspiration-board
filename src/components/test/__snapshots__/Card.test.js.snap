// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
<section
  className="card"
>
  <div
    className="card__content"
  >
    <div
      className="card__content-text"
    >
      text
    </div>
    <div
      className="card__content-emoji"
    />
    <button
      className="card__delete"
      onClick={[Function]}
    >
      DISCARD
    </button>
  </div>
</section>
`;
